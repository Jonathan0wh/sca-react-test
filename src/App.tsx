import React, { Fragment } from 'react';

import Header from 'Components/Header';
import SubscribeForm from 'Components/SubscribeForm';

const App = () => (
  <Fragment>
    <Header />
    <SubscribeForm />
  </Fragment>
);

export default App;
