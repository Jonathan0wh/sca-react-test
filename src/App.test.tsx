import React from 'react';
import { mount } from 'enzyme';

import App from './App';

it('render the whole App without crashing', () => {
  const wrapper = mount(<App />);

  expect(wrapper).toMatchSnapshot();
});
