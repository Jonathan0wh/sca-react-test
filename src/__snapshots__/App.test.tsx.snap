// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the whole App without crashing 1`] = `
<App>
  <Header>
    <header
      className="header"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <p
            className="text"
          >
            Subscribe
          </p>
        </div>
      </div>
    </header>
  </Header>
  <SubscribeForm>
    <form
      className="form"
    >
      <div
        className="container form_container"
      >
        <div
          className="row"
        >
          <div
            className="col-md-6"
          >
            <Input
              dataField="firstName"
              handleInputChange={[Function]}
              inputType="text"
              labelName="First Name"
              required={true}
            >
              <div
                className="input_wrapper"
              >
                <label
                  className="label_text"
                  htmlFor="input-First Name"
                >
                  First Name
                   
                  <span
                    className="asterisk"
                  >
                    *
                  </span>
                </label>
                <input
                  className="input"
                  id="input-First Name"
                  onChange={[Function]}
                  required={true}
                  type="text"
                />
              </div>
            </Input>
          </div>
          <div
            className="col-md-6"
          >
            <Input
              dataField="lastName"
              handleInputChange={[Function]}
              inputType="text"
              labelName="Last Name"
              required={true}
            >
              <div
                className="input_wrapper"
              >
                <label
                  className="label_text"
                  htmlFor="input-Last Name"
                >
                  Last Name
                   
                  <span
                    className="asterisk"
                  >
                    *
                  </span>
                </label>
                <input
                  className="input"
                  id="input-Last Name"
                  onChange={[Function]}
                  required={true}
                  type="text"
                />
              </div>
            </Input>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-6"
          >
            <Input
              dataField="email"
              handleInputChange={[Function]}
              inputType="email"
              labelName="Email"
              required={true}
            >
              <div
                className="input_wrapper"
              >
                <label
                  className="label_text"
                  htmlFor="input-Email"
                >
                  Email
                   
                  <span
                    className="asterisk"
                  >
                    *
                  </span>
                </label>
                <input
                  className="input"
                  id="input-Email"
                  onChange={[Function]}
                  required={true}
                  type="email"
                />
              </div>
            </Input>
          </div>
          <div
            className="col-md-6"
          >
            <Input
              dataField="mobilePhone"
              handleInputChange={[Function]}
              inputType="number"
              labelName="Mobile Phone"
              required={false}
            >
              <div
                className="input_wrapper"
              >
                <label
                  className="label_text"
                  htmlFor="input-Mobile Phone"
                >
                  Mobile Phone
                   
                </label>
                <input
                  className="input"
                  id="input-Mobile Phone"
                  onChange={[Function]}
                  required={false}
                  type="number"
                />
              </div>
            </Input>
          </div>
        </div>
        <div
          className="row"
        >
          <Button
            buttonText="Submit"
            handleClick={[Function]}
            loading={false}
          >
            <button
              className="button"
              onClick={[Function]}
            >
              Submit
            </button>
          </Button>
        </div>
      </div>
    </form>
  </SubscribeForm>
</App>
`;
